controller/src/main/java/io/pravega/controller/store/stream/records/SealedSegmentsMapShard.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream.records;
+
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.shared.segment.StreamSegmentNameUtils;
+import lombok.Builder;
+import lombok.Data;
+import lombok.SneakyThrows;
+import lombok.Synchronized;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Data class for storing information about stream's truncation point.
+ */
+@Builder
+@Slf4j
+@Data
+/**
+ * Sealed Segments Map is divided into multiple shards with each shard containing sealed sizes
+ * for a group of segments, with shard membership determined by a hash function. For example,
+ * we can use a simple modulo of the segment number to determine the shard where the
+ * corresponding sizes of the sealed segments are stored. An alternative is to take the epoch
+ * from the segmentId to compute the shard number.
+ *
+ * Each shard can contain unbounded number of segment ids, but if we use a good hash function,
+ * then the load is expected to be balanced across shards.
+*/
+public class SealedSegmentsMapShard {
+    public static final SealedSegmentsMapShardSerializer SERIALIZER = new SealedSegmentsMapShardSerializer();
+    public static final int SHARD_SIZE = 10000;
+
+    private final int shardNumber;
+    /**
+     * Sealed segments with size at the time of sealing.
+     * segmentId -> sealed segment record.
+     * Each shard contains segments from a range of `segment epoch`.
+     * So each shard size would be say 10k segment numbers. Then the number of records in the map would be 
```

Comment:
I don't understand what this is saying. The map is the union of all the shards, yes? The total number of records in the map should be the average number of segments per epoch x number of shards x number of epochs per shard.

My interpretation is probably wrong, but it reflects my confusion after reading the comment. I think this comment needs to be reworked.